lib/puppet/provider/foreman_smartproxy/rest_v2.rb

```diff
@@ -2,6 +2,12 @@
 
   confine :feature => :apipie_bindings
 
+  def raise_error(e)
+    body = JSON.parse(e.response)["error"]["full_messages"].join(" ") rescue 'N/A'
+    raise("Error while registering proxy (#{e.message}): #{body}, " +
+      "registration can be skipped by unsetting --register-in-foreman option.")
```

Comment:
Could we emphasise finding the reason before disabling it too, so people don't follow this advice and get a partially functioning installation?  IIRC we have debug output from apipie-bindings nowadays.